### PR TITLE
remove sessionId from required fields (because ELPAC won't have it)

### PR DESCRIPTION
**Pull request by malaffoon**
also bump version to 1.4.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ## Change Log
 
-#### 1.3.1
+#### 1.4.0 - 2019-
```
Comment (malaffoon) [inline CHANGELOG.md]: per comment elsewhere, let's bump this release to 1.4.0, it's going to be a doozy

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorIT.java [hunk 2]
```diff
@@ -51,10 +51,10 @@ public void itShouldProcessMessageWithXmlContent() {
         int beforeIabExamCount = countRowsInTableWhere(jdbcTemplate, "exam", "type_id = 2");
         int beforeOtherExamCount = countRowsInTableWhere(jdbcTemplate, "exam", "type_id <> 2");
         int beforeClaimScoreCount = countRowsInTable(jdbcTemplate, "exam_claim_score");
-        int beforeStudentCount = countRowsInTable(jdbcTemplate, "student");
-        int beforeStudentEthnicityCount = countRowsInTable(jdbcTemplate, "student_ethnicity");
-        int beforeGroupCount = countRowsInTable(jdbcTemplate, "student_group");
-        int beforeGroupMembershipCount = countRowsInTable(jdbcTemplate, "student_group_membership");
+        final int beforeStudentCount = countRowsInTable(jdbcTemplate, "student");
```
Comment (malaffoon) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorIT.java]: unrelated, fix warnings

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorIT.java [hunk 3]
```diff
@@ -160,18 +159,17 @@ public void itShouldReturnErrorsForDefaultRequiredStudentDataElement() {
...
         try {
             final TDSReport tdsReport = XmlUtils.tdsReportFromXml(this.getClass().getResourceAsStream("/NULL-fields.xml"));
             processor.process(tdsReport, importId);
             fail("it should fail");
-        } catch (ImportException ex) {
+        } catch (final ImportException ex) {
             assertThat(ex.getMessage()).isEqualTo("{\"messages\":[" +
                     "{\"elementName\":\"IDEAIndicator\",\"error\":\"invalid value [null]\"}," +
-                    "{\"elementName\":\"EconomicDisadvantageStatus\",\"error\":\"invalid value [null]\"}," +
+                    "{\"elementName\":\"EconomicDisadvantageStatus\",\"error\":\"invalid value [null]\"}" +
```
Comment (malaffoon) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorIT.java]: the actual change to the test

File: exam-processor/src/main/resources/application.yml [hunk 4]
```diff
@@ -69,6 +69,5 @@ validation:
     - Sex
     - EconomicDisadvantageStatus
     - IDEAIndicator
-    - SessionId
```
Comment (malaffoon) [inline exam-processor/src/main/resources/application.yml]: the actual change

File: CHANGELOG.md (hunk 1 above)
Comment (esotrope) [reply CHANGELOG.md]: 👍

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
...
+
+* Make sessionId optional by default (change `validation.requiredDataElements`)
```
Comment (esotrope) [inline CHANGELOG.md]: so is the idea later to introduce a flag to control whether or not this check is made at ingest time?

Comment (malaffoon) [reply CHANGELOG.md]: There is already a flag to control it. This just turns it off by default. I added the note here so installations can check to see if they have a related configuration that might need to be adjusted (the CA installation had a habit of copying the entire yml configuration, so they will have to make this same change in their system).

**Pull request merged**